externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala

```diff
@@ -295,13 +295,19 @@ object SparkDatasetHelper extends Logging {
     SQLMetrics.postDriverMetricUpdates(sc, executionId, metrics.values.toSeq)
   }
 
+  private[kyuubi] def logicalPlanLimit(plan: LogicalPlan): Option[Long] = plan match {
+    case globalLimit: GlobalLimit => globalLimit.maxRows
+    case _ => None
+  }
+
   def shouldSaveResultToFs(resultMaxRows: Int, minSize: Long, result: DataFrame): Boolean = {
     if (isCommandExec(result.queryExecution.executedPlan.nodeName)) {
       return false
     }
-    lazy val limit = result.queryExecution.executedPlan match {
-      case collectLimit: CollectLimitExec => collectLimit.limit
-      case _ => resultMaxRows
+    val limit = logicalPlanLimit(result.queryExecution.logical) match {
```

Comment:
can we have a different name for this variable? e.g. `finalLimit`, `effectiveLimit`, to distinguish with the `Some(limit)`